site/src/pages/WorkspaceSchedulePage/WorkspaceSchedulePage.tsx

```diff
@@ -111,16 +113,40 @@ export const WorkspaceSchedulePage: React.FC = () => {
             type: "SUBMIT_SCHEDULE",
             autoStart: formValuesToAutoStartRequest(values),
             ttl: formValuesToTTLRequest(values),
-            autoStartChanged: scheduleChanged(autoStart, values),
-            autoStopChanged: scheduleChanged(autoStop, values),
+            autoStartChanged: scheduleChanged(getAutoStart(workspace), values),
+            autoStopChanged: scheduleChanged(getAutoStop(workspace), values),
           })
         }}
       />
     )
   }
 
-  if (scheduleState.matches("submitSuccess")) {
-    return <Navigate to={`/@${username}/${workspaceName}`} />
+  if (scheduleState.matches("showingRestartDialog")) {
+    return (
+      <ConfirmDialog
+        open
+        title={Language.dialogTitle}
+        description={Language.dialogDescription}
+        confirmText={Language.restart}
+        cancelText={Language.applyLater}
+        hideCancel={false}
+        onConfirm={() => {
+          scheduleSend("RESTART_WORKSPACE")
+        }}
+        onClose={() => {
+          scheduleSend("APPLY_LATER")
+        }}
+      />
+    )
+  }
+
+  if (scheduleState.matches("done")) {
+    return (
+      <Navigate
+        to={`/@${username}/${workspaceName}`}
+        state={{ shouldRestartWorkspace }}
```

Comment:
I'm surprised to see us passing this as part of `Location`. I would only expect to see routing-specific variables in that object.

Is there a reason we can't post a new build for the workspace from inside of the workspaceSchedule machine after the TTL post returns, then navigate to the workspace page?

Reply:
Since starting a workspace takes several seconds, and since the workspace page already handles errors for it, I wanted to go back to the workspace page first and then make the api call. But since our workspace X service is only alive when the page is mounted (which we could change, but I was afraid that would be too big a change), I can't send it a message from the schedule page. Maybe I could fire and forget from the schedule page since the workspace page gets updates pushed from the server?

Reply:
Yeah, good idea: I think you could fire and forget! I think the only waiting we need to do is ensuring that TTL post finishes first, so we don't restart the workspace if something goes awry.